src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -798,12 +711,41 @@ public int getCurrentPoolSize() {
      * @return current number of workers executing tasks
      */
     public int getCurrentRunningCount() {
-      synchronized (workersLock) {
-        int workerOutQty = currentPoolSize - availableWorkers.size();
-        if (waitingForWorker || currentPoolSize == 0) {
-          return workerOutQty;
+      int startPoolSize = currentPoolSize.get();
+      int initialCount = 0;
+      int initialChainValue = CHAIN_VERIFICATION_START_VAL;
+      while (true) {
+        // TODO - in threadly 5.0.0 we can make this simpler
+        if (initialCount == 0) {
+          Worker idleWorker = this.idleWorker.get();
+          while (idleWorker != null) {
+            initialCount++;
+            initialChainValue ^= idleWorker.hashCode();
+            idleWorker = idleWorker.nextIdleWorker;
+          }
+        }
+        int verificationCount = 0;
+        int verificationChainValue = CHAIN_VERIFICATION_START_VAL;
+        Worker idleWorker = this.idleWorker.get();
+        while (idleWorker != null) {
```

Comment:
I would pull this count logic into its own private function, something like getLazyIdleCount().